### PR TITLE
Feat/nearest neighbor

### PR DESCRIPTION
**Pull request by ynydara**
gets the nearest neighbors for knn algorithm

### CODE REVIEW

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighborAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.github.gissuite.gribinterpolation.core;
+
+import com.github.gissuite.gribinterpolation.data.DataPoint;
+
+import java.util.Arrays;
+
+import static com.github.gissuite.gribinterpolation.core.NearestNeighbor.getNearestNeighbor;
+
+public class NearestNeighborAlgorithm {
```
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighborAlgorithm.java:9]: This whole file shouldn't exist

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.github.gissuite.gribinterpolation.core;
+import com.github.gissuite.gribinterpolation.data.DataPoint;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class NearestNeighbor {
```
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:7]: add a line after the class

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighborAlgorithm.java (hunk 1 above)
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighborAlgorithm.java]: This whole file shouldn't exist

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+private static Map<DataPoint, Double> sortByValue(HashMap<DataPoint, Double> map) {
```
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:8]: Private methods should go below public ones

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    return map.entrySet()
+            .stream()
+            .sorted(Map.Entry.comparingByValue())
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (firstValue, streamValue) -> firstValue, LinkedHashMap::new));
+}
+    public static DataPoint[] getNearestNeighbor(DataPoint[] dataPoint, float longitudeToInterpolate, float latitudeToInterpolate, int k, int amountOfDataPoints) {
```
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:14]: Please create proper documentation for this method

Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:14]: the `dataPoint` parameter should be plural because it's an array

Comment (sechols223) [reply src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:14]: the `dataPoint` parameter should be an ArrayList

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        Map<DataPoint, Double> sortedHashMap = sortByValue(hashMap);
+        Set<DataPoint> keySet = sortedHashMap.keySet();
+        Iterator<DataPoint> iterator = keySet.iterator();
+        int count = 0;
+        while(iterator.hasNext() && count < k){
+            DataPoint key = iterator.next();
+            nearestNeighbors[count] = key;
+            count++;
+        }
+        return nearestNeighbors;
```
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:30]: The output of this method should be an ArrayList

**Review by sechols223 — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java (hunk 5 above)
Comment (ynydara) [reply src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java]: ok done

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java (hunk 4 above)
Comment (ynydara) [reply src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java]: done

Comment (ynydara) [reply src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java]: hopefully done :)

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java (hunk 3 above)
Comment (ynydara) [reply src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:36]: done

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighborAlgorithm.java (hunk 1 above)
Comment (ynydara) [reply src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighborAlgorithm.java]: deleted

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java (hunk 2 above)
Comment (ynydara) [reply src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:15]: done

File: src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+//gets k nearest neighbors' memory locations stored in a DataPoint array
+//amountOfDataPoints is how many points to find the distance for; so should be how many DataPoints in total we have.
+//k is how many neighbors to look for
+
+//can get temperature this way:
+//ArrayList<DataPoint> nearestNeighbors = (getNearestNeighbor(arrayOfDataPoints, longitudeToInterpolate, latitudeToInterpolate, k, amountOfDataPoints));
+//DataPoint a = (DataPoint) nearestNeighbors.get(0);
+//System.out.println(a.getTemperatureK());
+public class NearestNeighbor {
```
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/NearestNeighbor.java:15]: Use this as a reference for creating proper documentation using Javadoc:
https://www.baeldung.com/javadochttps://www.baeldung.com/javadoc

**Review by sechols223 — CHANGES_REQUESTED** [review]

**Review by sechols223 — APPROVED** [review]

**Pull request merged**